instrumentation/http/src/main/java/brave/http/HttpResponse.java

```diff
@@ -28,31 +28,45 @@
  */
 public abstract class HttpResponse extends Response {
   /**
-   * Like {@link HttpRequest#method()} except used in response parsing.
+   * The request that initiated this HTTP response or {@code null} if unknown.
    *
-   * <p>Notably, this is used to create a route-based span name.
+   * <p>Implementations should return the last wire-level request that caused this response or
+   * error. HTTP properties like {@linkplain HttpRequest#path() path} and {@linkplain
+   * HttpRequest#header(String) headers} might be different, due to redirects or authentication.
+   * Some properties might not be visible until response processing, notably {@link #route()}.
    *
    * @since 5.10
    */
-  // TODO: Consider `httpResponse.request()` and deprecating `httpResponse.method()` #1086
-  @Nullable public String method() {
+  @Nullable public HttpRequest request() {
     return null;
   }
 
   /**
-   * Returns an expression such as "/items/:itemId" representing an application endpoint,
-   * conventionally associated with the tag key "http.route". If no route matched, "" (empty string)
-   * is returned. Null indicates this instrumentation doesn't understand http routes.
+   * Returns the {@linkplain HttpRequest#method() HTTP method} of the request that caused this
+   * response or {@code null} if unreadable.
```

Comment:
Maybe it's worth adding Where possible, prefer to populate `{@link #request()} to provide both the method and other information that may be useful during response parsing.

Reply:
sure